singularity/local.go

```diff
@@ -0,0 +1,163 @@
+package singularity
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/data-preservation-programs/singularity/database"
+	"github.com/data-preservation-programs/singularity/datasource"
+	"github.com/data-preservation-programs/singularity/model"
+	"github.com/data-preservation-programs/singularity/service/datasetworker"
+	fs2 "github.com/rclone/rclone/fs"
+	"gorm.io/gorm"
+)
+
+var (
+	ErrItemNotFound   = errors.New("no item is found with the given id")
+	ErrSourceNotFound = errors.New("no source is found with the given id")
+	ErrNotInitialized = errors.New("singularity source not initialized")
+)
+
+const MOTION_DATASET_NAME = "MOTION_DATASET"
+const PIECE_SIZE = 1 << 36
+const MAX_CAR_SIZE = (1 << 29) * 127
+
+type LocalSingularity struct {
+	db                        *gorm.DB
+	datasourceHandlerResolver datasource.HandlerResolver
+	initialized               bool
+	sourceID                  uint32
+}
+
+func OpenSQLiteSingularity(dbFile string) (*LocalSingularity, error) {
+	db, err := database.Open("sqlite:"+dbFile, &gorm.Config{})
```

Comment:
Do you want to use this?
https://github.com/data-preservation-programs/singularity/blob/9e60f0b83c3dc29af1906402a8bd2f433b1e6780/database/connstring.go#L21
So that you can just use a single connection string for all supported backends
https://github.com/data-preservation-programs/singularity/blob/9e60f0b83c3dc29af1906402a8bd2f433b1e6780/cmd/app.go#L34

It also includes some pragmas that enables foreign key support in SQlite